x/ibc/20-transfer/keeper/keeper.go

```diff
@@ -72,11 +76,22 @@ func (k Keeper) PacketExecuted(ctx sdk.Context, packet channelexported.PacketI,
 // ChanCloseInit defines a wrapper function for the channel Keeper's function
 // in order to expose it to the ICS20 trasfer handler.
 func (k Keeper) ChanCloseInit(ctx sdk.Context, portID, channelID string) error {
-	return k.channelKeeper.ChanCloseInit(ctx, portID, channelID)
+	capName := ibctypes.ChannelCapabilityPath(portID, channelID)
+	chanCap, ok := k.scopedKeeper.GetCapability(ctx, capName)
+	if !ok {
+		return sdkerrors.Wrapf(channel.ErrChannelCapabilityNotFound, "could not retrieve channel capability at: %s", capName)
+	}
+	return k.channelKeeper.ChanCloseInit(ctx, portID, channelID, chanCap)
 }
 
 // TimeoutExecuted defines a wrapper function for the channel Keeper's function
 // in order to expose it to the ICS20 transfer handler.
 func (k Keeper) TimeoutExecuted(ctx sdk.Context, packet channelexported.PacketI) error {
 	return k.channelKeeper.TimeoutExecuted(ctx, packet)
 }
+
+// ClaimCapability allows the transfer module that can claim a capability that IBC module
+// passes to it
+func (k Keeper) ClaimCapability(ctx sdk.Context, cap capability.Capability, name string) error {
```

Comment:
Should this be a public function, though? Where are we using this at the moment?

Reply:
We use it in `20-transfer/module.go` since the callbacks must `ClaimCapability` and are not part of the keeper package

Reply:
Perhaps you might be able to call ClaimCapability with a known portID/channelID and a new capability to "replace" the capability and remove the module's ability to send packets if you have access to the keeper....

Reply:
Is that a valid attack we should be worried about?

Reply:
That is a potential concern, maybe this function should only be exposed to the IBC keeper.

Reply:
Looked through the capability code. This isn't a possible attack, because it is not possible to "replace" a capability with the same name:

https://github.com/cosmos/cosmos-sdk/blob/ibc-alpha/x/capability/keeper/keeper.go#L129

Tried to work out the cleanest way to do this and there wasn't a natural way to do it. Given this isn't an actual risk, believe we should merge as-is

Reply:
OK